src/main/java/seedu/taassist/logic/commands/DeletecCommand.java

```diff
@@ -45,14 +52,20 @@ public CommandResult execute(Model model) throws CommandException {
                     model.getModuleClassList()));
         }
 
-        List<Student> students = model.getFilteredStudentList();
+        List<Student> students = model.getTaAssist().getStudentList();
         for (Student student : students) {
             if (!Collections.disjoint(student.getModuleClasses(), moduleClasses)) {
                 Set<ModuleClass> assignedClasses = new HashSet<>();
                 assignedClasses.addAll(student.getModuleClasses());
                 assignedClasses.removeAll(moduleClasses);
+
+                List<StudentModuleData> studentModuleData = student.getModuleDataList();
+                List<StudentModuleData> updatedModuleData = studentModuleData.stream()
+                        .filter(data -> !assignedClasses.contains(data.getModuleClass()))
```

Comment:
Using `contains` is correct for now. However, since `ModuleClass` should use `isSame` to check for equality, this might be an issue in the future, will KIV in a future issue.

Reply:
Mentioned in new issue #152.